### PR TITLE
Upgrade template

### PR DESCRIPTION
**Pull request by pacrob**
### What was wrong?

Template updates, notably dropping python 3.7, using `pre-commit` for linting, and change `master` branch to `main`.

### Todo:
- [x] Clean up commit history

- [x] Add or update documentation related to these changes

- [x] Add entry to the [release notes](https://github.com/ethereum/eth-account/blob/master/newsfragments/README.md)

#### Cute Animal Picture

image

### CODE REVIEW

File: 248.breaking.rst [hunk 1]
```diff
@@ -0,0 +1 @@
+Drop support for python 3.7
```
Comment (reedsa) [inline 248.breaking.rst]: Noticed these are placed outside of the `newsfragments` dir.

Comment (pacrob) [reply 248.breaking.rst]: oof! good catch, thank you!

**Review by reedsa — APPROVED** [review]
lgtm

**Review by kclowes — APPROVED** [review]
🎉 !

**Pull request merged**